lib/blacklight/path.rb

```diff
@@ -0,0 +1,131 @@
+module Blacklight
+  # This class encapsulates the search state as represented by the query
+  # parameters namely: :f, :q, :page, :per_page and, :sort
+  class Path
+    include Blacklight::Facet
```

Comment:
Yes, I believe that is all.  I agree that belongs in the configuration. I don't want to keep mixing this in.